application/src/main/java/org/openhds/mobile/activity/NavigateActivity.java

```diff
@@ -685,6 +760,8 @@ public void onEnterState() {
                     validForms.add(form);
                 }
             }
+            deleteSubFormPaths();
+            populateFormsForPath();
```

Comment:
Since we're calling these together in multiple places, it's probably a good candidate for factoring into a single method. Perhaps called something like, updateAssociatedForms()?